hardly/handlers/distgit.py

```diff
@@ -51,6 +51,32 @@ def __init__(
         )
         self.target_repo_branch = event["target_repo_branch"]
 
+    def get_dist_git_branch(self) -> str:
+        """
+        Get the downstream repo branch name where to open the MR.
+
+        If the downstream repo branch does not exist
+        the upstream source git PR will be commented
+        to communicate to the user that a new branch has
+        been created.
+
+        Returns:
+            The downstream repo branch name
+        """
+        dist_git_branch = (
+            self.package_config.downstream_branch_name
+            if self.package_config.downstream_branch_name
+            else self.target_repo_branch
+        )
+        if dist_git_branch not in self.api.dg.local_project.git_project.get_branches():
+            msg = f"""
+{self.target_repo}:{dist_git_branch} does not exist.
+A new branch with this name will be created into the downstream repo.
+"""
+            self.project.get_pr(int(self.mr_identifier)).comment(msg)
+
+        return dist_git_branch
```

Comment:
Tbh, I wouldn't do this check here. If the branch doesn't exist in dist-git, there is nothing meaningful the bot could do: creating it is a release engineering task, it shouldn't be created on the fly. Another solution could be to set `downstream_branch_name`, but that's also up to the maintainer to decide.

So if the branch doesn't exist in dist-git, opening the PR should simply fail.